binderhub/builder.py

```diff
@@ -544,21 +582,30 @@ def _check_result(future):
             )
             BUILD_COUNT.labels(status="success", **self.repo_metric_labels).inc()
             with LAUNCHES_INPROGRESS.track_inprogress():
-                await self.launch(provider)
-            self.event_log.emit(
-                "binderhub.jupyter.org/launch",
-                5,
-                {
-                    "provider": provider.name,
-                    "spec": spec,
-                    "ref": ref,
-                    "status": "success",
-                    "build_token": self._have_build_token,
-                    "origin": self.settings["normalized_origin"]
-                    if self.settings["normalized_origin"]
-                    else self.request.host,
-                },
-            )
+                if build_only:
+                    await self.emit(
+                        {
+                            "phase": "ready",
+                            "imageName": image_name,
+                            "message": "Image won't be launched\n",
+                        }
+                    )
+                else:
+                    await self.launch(provider)
+                    self.event_log.emit(
+                        "binderhub.jupyter.org/launch",
+                        5,
+                        {
+                            "provider": provider.name,
+                            "spec": spec,
+                            "ref": ref,
+                            "status": "success",
+                            "build_token": self._have_build_token,
+                            "origin": self.settings["normalized_origin"]
+                            if self.settings["normalized_origin"]
+                            else self.request.host,
+                        },
+                    )
```

Comment:
I read about event logging in https://binderhub.readthedocs.io/en/latest/eventlogging.html. I figure it could be relevant to track the amount of builds we do, even though binderhub isn't launching.

I think that could be a related but separate PR though, independent on this feature.